discovery/discovery.go

```diff
@@ -221,7 +221,7 @@ func (d *discovery) checkCluster() ([]*client.Node, int, uint64, error) {
 		if err == client.ErrInvalidJSON {
 			return nil, 0, 0, ErrBadDiscoveryEndpoint
 		}
-		if err == context.DeadlineExceeded {
+		if _, ok := err.(*client.ClusterError); ok {
```

Comment:
I think your example is a bad one. How can you tell if the discovery is down or it cannot be found? 
After think a while, I guess we are doing the right thing now by retrying all network error from func do.